packages/amplify-category-analytics/src/plugin-client-api-notifications.ts

```diff
@@ -29,7 +29,8 @@ export const invokeNotificationsAPIRecursiveRemoveApp = async (
  * @param resourceName Pinpoint resource name
  * @returns true if Pinpoint resource is in use
  */
-export const checkResourceInUseByNotifications = async (context: $TSContext, resourceName: string): Promise<boolean> => {
+export const checkResourceInUseByNotifications = async (context: $TSContext, resourceName?: string): Promise<boolean> => {
+  if (!resourceName) return false;
```

Comment:
would it let write code like `checkResourceInUseByNotifications(context);` ? 
If so is there a different way to let null/undefined in? (why is this even a case?)

Reply:
I can gate this case upstream of this function

Reply:
that'd be cleaner too. I don't mind passing undef but I do mind if IDE hints wrong usage.